doc/user_guide/ensembleModel.tex

```diff
@@ -0,0 +1,229 @@
+ballistics, then impact) and transfer data from one to the next.
+
+The third data object, \xmlString{final\_results}, can contain any information we want it to.  In this case,
+we want to collect all the variables in their original spaces, so we take \texttt{y0,v0,ang,m,r} as inputs and
+\texttt{E,D} as outputs.
+
+
+\subsubsection{Files}
+Since we know both of our models take input files to set the variable values, we inform RAVEN about the
+template input files and
+give those files RAVEN names in the \xmlNode{Files} block.  In this example, our input file for \texttt{ballistics.py} is
+\texttt{ballistics\_template.txt}, which we simply name \xmlString{ballistics\_input}.  Similarly,
+for \texttt{impact.py} the input file is \texttt{impact\_template.txt}, which we name
+\xmlString{impact\_input}:
+\xmlExample{framework/user_guide/EnsembleModel/basic.xml}{Files}
+We will pause to note that the file \texttt{ballistics\_template.txt} is different than the file
+\texttt{ballistics\_input.py}, and similarly for the impact code.  In the template file, we've replaced
+variable values with the \texttt{\$RAVEN-\$} wildcard, since we will be using the Generic Code model for these
+two models.  Having a template input file may not be required for all code interfaces, but it is required for
+the Generic interface. See more about this interface in the Models section and in the user manual.
+
+
+\subsubsection{Models}
+Once we've specified the data objects and files, we are prepared to set up the \xmlNode{Models} block.  To set
+up an ensemble model, we define each sub-module by itself as if it were a stand-alone RAVEN model, and then
+combine them by defining an \xmlNode{EnsembleModel}:
+\xmlExample{framework/user_guide/EnsembleModel/basic.xml}{Models}
+The first model we will look at is the definition of the \xmlString{ballistics} model.  We note that is uses
+the \xmlString{GenericCode} subType, so it will use the interface as described in the user manual, where
+wildcards in the input file are replaced with RAVEN's sampled values.  The \xmlNode{clargs} (command line
+arguments) include the \xmlString{python} command, the \xmlString{-i} flag to specify the input file, and the
+\xmlString{-o} flag to specify the output file.  These nodes are specific to the Generic code interface and
+will not apply to all codes or models. The \xmlString{Impact} model is defined in a similar manner.
+
+With both Models defined, we can construct the \xmlNode{EnsembleModel}, which we've named
+\xmlString{ballistic\_and\_impact} to describe the codes that are paired (you can name this whatever you
+want).  There is no \xmlAttr{subType} for the \xmlNode{EnsembleModel} currently, so we leave it blank.
+
+Within the \xmlNode{EnsembleModel} node there are two \xmlNode{Model} nodes listed, which will inform the
+Ensemble Model which models need coupling.  The order listed does not matter; the Ensemble Model will sort out
```

Comment:
```EnsembleModel```

Reply:
Can Ensemble Model only be spelled like it is in the code, or can it be used in a sentence?

Reply:
it is just to keep a link with the entity in the RAVEN input.